packages/framework/src/Support/Filesystem/RelativePathString.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Hyde\Support\Filesystem;
+
+use Hyde\Hyde;
+use JetBrains\PhpStorm\ArrayShape;
+
+/**
+ * Denotes and enforces a path relative to the project root.
+ */
+final class RelativePathString extends PathString
```

Comment:
Since all paths used by Hyde are (should) be relative up until the time they are used to actually interact with the file system should be relative. We only convert it to an absolute Hyde::path() before loading/saving a file. The idea behind the path-string type system is to ensure we relative paths, but that might better be onforced by only having an absolute path string type, as everything else is expected to be relative. That also better conveys the difference I think.